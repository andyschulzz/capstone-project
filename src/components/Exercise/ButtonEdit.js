import React from 'react'
import Button from '../common/Button'
import { Link } from 'react-router-dom'

export default function ButtonEdit() {
  return (
    <Button mla="true" as={Link} to={`/exercises/edit`}>
      Edit
    </Button>
  )
}
