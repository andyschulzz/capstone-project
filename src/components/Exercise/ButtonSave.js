import React from 'react'
import { GreenButton } from '../common/Button'

export default function ButtonSave({ formState, form }) {
  return (
    <>
      {formState.isValid ? (
        <GreenButton mla type="submit" form={form}>
          Save
        </GreenButton>
      ) : (
        <GreenButton inactive mla type="submit" form={form}>
          Save
        </GreenButton>
      )}
    </>
  )
}
