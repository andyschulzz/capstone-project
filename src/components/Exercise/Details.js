import React, { useEffect, useRef, useState } from 'react'
import { useForm } from 'react-hook-form'
import ButtonBackToHome from '../common/ButtonBackToHome'
import ButtonEdit from './ButtonEdit'
import Form from './Form'
import * as S from './Form.styles'
import PropTypes from 'prop-types'
import { useParams } from 'react-router-dom'
import { BlueButton, GreenButton } from '../common/Button'

Details.propTypes = {
  handleExerciseChange: PropTypes.func.isRequired,
  exercises: PropTypes.array.isRequired,
  name: PropTypes.string,
  type: PropTypes.string,
  instructions: PropTypes.string,
  image: PropTypes.string,
}

export default function Details({ handleExerciseChange, exercises }) {
  const { id } = useParams()
  const { register, setValue, handleSubmit } = useForm()
  const [disabled, setDisabled] = useState(true)
  const [image, setImage] = useState()
  const mainRef = useRef(null)

  const selectedExercise = exercises.filter((exercise) => exercise.id === id)

  useEffect(() => {
    selectedExercise.forEach((exercise) => {
      setValue('name', exercise.name)
      setValue('type', exercise.type)
      setValue('instructions', exercise.instructions)
      setImage(exercise.image)
    })
  }, [selectedExercise, setValue])

  useEffect(() => {
    mainRef.current.focus()
  }, [disabled])

  return (
    <S.Form
      onSubmit={handleSubmit(handleChange)}
      data-testid="details"
      id="details"
    >
      <S.ButtonWrapper>
        <ButtonBackToHome />
        {(!disabled && (
          <GreenButton inactive mla>
            Save
          </GreenButton>
        )) || (
          <BlueButton mla onClick={handleChange}>
            Edit
          </BlueButton>
        )}
      </S.ButtonWrapper>
      <Form
        mainRef={mainRef}
        register={register}
        image={image}
        disabled={disabled}
      />
    </S.Form>
  )

  function handleChange(data) {
    setDisabled(!disabled)
    if (!disabled) {
      handleExerciseChange({ id, ...data })
    }
  }
}
