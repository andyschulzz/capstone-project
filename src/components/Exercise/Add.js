import React, { useState } from 'react'
import { useForm } from 'react-hook-form'
import Form from './Form'
import ButtonBackToHome from '../common/ButtonBackToHome'
import ButtonEdit from './ButtonEdit'
import ButtonSave from './ButtonSave'
import * as S from './Form.styles'
import PropTypes from 'prop-types'

Add.propTypes = {
  handleExerciseAdd: PropTypes.func.isRequired,
}

export default function Add({ handleExerciseAdd }) {
  const [disabled, setDisabled] = useState(false)
  const { register, handleSubmit, formState } = useForm({ mode: 'onChange' })

  return (
    <>
      <S.ButtonWrapper>
        <ButtonBackToHome />
        {(!disabled && <ButtonSave formState={formState} form={'add'} />) || (
          <ButtonEdit />
        )}
      </S.ButtonWrapper>
      <S.Wrapper id="add" onSubmit={handleSubmit(handleAdd)}>
        <Form register={register} disabled={disabled} />
      </S.Wrapper>
    </>
  )
  function handleAdd(data) {
    Object.keys(data).map(key => (data[key] = data[key].trim()))
    setDisabled(!disabled)
    if (!disabled) {
      handleExerciseAdd(data.name, data.type, data.instructions)
    }
  }
}
