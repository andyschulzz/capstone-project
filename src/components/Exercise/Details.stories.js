import React from 'react'
import Details from './Details'

export default {
  title: 'Exercise/Details',
  component: Details,
  decorators: [
    renderComponent => (
      <div style={{ padding: 20, width: 400 }}>{renderComponent()}</div>
    ),
  ],
}

export const Default = () => {
  return (
    <Details
      name={'Example Name'}
      type={'Example Type'}
      instructions={'Example Instructions'}
    />
  )
}
