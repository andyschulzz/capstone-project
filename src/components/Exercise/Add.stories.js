import React from 'react'
import Add from './Add'

export default {
  title: 'Exercise/Add',
  component: Add,
  decorators: [
    (renderComponent) => (
      <div style={{ padding: 20, width: 400 }}>{renderComponent()}</div>
    ),
  ],
}

export const Default = () => {
  return <Add />
}
