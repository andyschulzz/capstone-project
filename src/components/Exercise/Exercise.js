import React from 'react'
import * as S from './Exercise.styles'
import { useRouteMatch } from 'react-router-dom'
import PropTypes from 'prop-types'

Exercise.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleExerciseSelect: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
}

export default function Exercise({
  name,
  type,
  handleExerciseSelect,
  id,
  index,
}) {
  const { url } = useRouteMatch()
  return (
    <S.Exercise
      to={`${url}/details/${index + 1}`}
      onClick={() => handleExerciseSelect(id)}
    >
      <S.Text isName>{name}</S.Text>
      <S.Text>{type}</S.Text>
    </S.Exercise>
  )
}
