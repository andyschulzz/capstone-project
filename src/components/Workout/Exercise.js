import React from 'react'
import * as S from './Exercise.styles'
import PropTypes from 'prop-types'
import { useToggle } from 'react-hooks-lib'

Exercise.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleWorkoutExercises: PropTypes.func,
  id: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
}

export default function Exercise({
  name,
  type,
  handleWorkoutExercises,
  id,
  image,
}) {
  const { on, toggle } = useToggle(false)
  return (
    <S.Exercise data-testid="exercise" onClick={() => handleWorkoutSelect()}>
      {on ? (
        <S.Image
          data-testid="checked"
          isChecked
          src="/icons/checked.png"
          alt=""
        />
      ) : (
        <S.Image src={image} alt="" />
      )}
      <S.Wrapper>
        <S.Text isName>{name}</S.Text>
        <S.Text isType>{type}</S.Text>
      </S.Wrapper>
    </S.Exercise>
  )
  function handleWorkoutSelect() {
    handleWorkoutExercises(id)
    toggle()
  }
}
