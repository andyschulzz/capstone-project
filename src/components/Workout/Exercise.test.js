import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Exercise from './Exercise.js'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)
describe('<Exercise />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <Exercise name="Arnold Press" type="Shoulder" />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
