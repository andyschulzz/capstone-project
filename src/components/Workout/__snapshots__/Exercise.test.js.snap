// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exercise /> should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 18px 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c0 p:first-child {
  margin-top: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 p:first-child {
  margin-top: 0;
}

.c3 {
  font-size: 1rem;
  margin: 10px 0 0 0;
}

.c4 {
  font-size: 0.8rem;
  margin: 0;
  font-weight: 300;
  color: #999;
}

.c1 {
  width: 46px;
  height: 46px;
  margin: 0 36px 0 0;
}

<div
    class="c0"
    data-testid="exercise"
  >
    <img
      alt=""
      class="c1"
      src="placeholder.png"
    />
    <div
      class="c2"
    >
      <p
        class="c3"
      >
        Arnold Press
      </p>
      <p
        class="c4"
      >
        Shoulder
      </p>
    </div>
  </div>
</DocumentFragment>
`;
